docs/src/resources/two-side-column.md

```diff
@@ -0,0 +1,519 @@
+  input  [7:0] io_in_bits,
+  input        io_out_ready,
+  output       io_out_valid,
+  output [7:0] io_out_bits
+);
+  assign io_in_ready = io_out_ready; // @[main.scala 17:12]
+  assign io_out_valid = io_in_valid; // @[main.scala 17:12]
+  assign io_out_bits = io_in_bits; // @[main.scala 17:12]
+endmodule
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+#Register assignment
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+            <td><b style="font-size:30px">Generated Verilog</b></td>
+         </tr>
+         <tr>
+<td>Text comes here</td>
+<td>
+
+```scala mdoc:silent
+class RegisterModule extends Module {
+  val io = IO(new Bundle {
+    val in  = Input(UInt(12.W))
+    val out = Output(UInt(12.W))
+  })
+
+val register = Reg(UInt(12.W))
+register := io.in + 1.U
+io.out := register
+}
+
+println(getVerilogString(new RegisterModule))
+```
+</td>
+<td>
+
+```
+module RegisterModule(
+  input         clock,
+  input         reset,
+  input  [11:0] io_in,
+  output [11:0] io_out
+);
+`ifdef RANDOMIZE_REG_INIT
+  reg [31:0] _RAND_0;
+`endif // RANDOMIZE_REG_INIT
+  reg [11:0] register; // @[main.scala 14:21]
+  assign io_out = register; // @[main.scala 16:10]
+  always @(posedge clock) begin
+    register <= io_in + 12'h1; // @[main.scala 15:21]
+  end
+// Register and memory initialization
+`ifdef RANDOMIZE_GARBAGE_ASSIGN
+`define RANDOMIZE
+`endif
+`ifdef RANDOMIZE_INVALID_ASSIGN
+`define RANDOMIZE
+`endif
+`ifdef RANDOMIZE_REG_INIT
+`define RANDOMIZE
+`endif
+`ifdef RANDOMIZE_MEM_INIT
+`define RANDOMIZE
+`endif
+`ifndef RANDOM
+`define RANDOM $random
+`endif
+`ifdef RANDOMIZE_MEM_INIT
+  integer initvar;
+`endif
+`ifndef SYNTHESIS
+`ifdef FIRRTL_BEFORE_INITIAL
+`FIRRTL_BEFORE_INITIAL
+`endif
+initial begin
+  `ifdef RANDOMIZE
+    `ifdef INIT_RANDOM
+      `INIT_RANDOM
+    `endif
+    `ifndef VERILATOR
+      `ifdef RANDOMIZE_DELAY
+        #`RANDOMIZE_DELAY begin end
+      `else
+        #0.002 begin end
+      `endif
+    `endif
+`ifdef RANDOMIZE_REG_INIT
+  _RAND_0 = {1{`RANDOM}};
+  register = _RAND_0[11:0];
+`endif // RANDOMIZE_REG_INIT
+  `endif // RANDOMIZE
+end // initial
+`ifdef FIRRTL_AFTER_INITIAL
+`FIRRTL_AFTER_INITIAL
+`endif
+`endif // SYNTHESIS
+endmodule
+```
+</td>
+         </tr>
+    </table>
+<html>
+<body>
+
+#Case statement
+
+<html>
+<body>
+    <table border ="0">
+          <tr>
+            <td><b style="font-size:30px">Verilog</b></td>
+            <td><b style="font-size:30px">Chisel</b></td>
+            <td><b style="font-size:30px">Generated Verilog</b></td>
+         </tr>
+         <tr>
+<td>Text comes here</td>
+<td>
+
+```scala mdoc:silent
+class DelayBy1(resetValue: Option[UInt] = None) extends Module {
+  val io = IO(new Bundle {
+    val in  = Input( UInt(16.W))
+    val out = Output(UInt(16.W))
+  })
+  val reg = resetValue match {
```

Comment:
this isn't quite the right example. This is another example where this is a "circuit generation time" decision -- the `match` is saying "what circuit do we want to build here"? We want to have a "circuit execution time" decision -- we need the construct that makes the circuit put down enough wires and logic to compute the decision once electricit is actually flowing on the wires on the chip.

For Chisel, that is the `switch` construct which is on the Cheat Sheet. For verilog there is the `case ` statement.

We can use the `module_mux` example here and translate it to Chisel (annoyingly they make it hard to copy the text directly)
https://www.chipverify.com/verilog/verilog-case-statement

Reply:
while working on this... 
I am stuck as to how to complete the code:
here is the scastie link: https://scastie.scala-lang.org/Shorla/o1ShdaY3RWKf0IIFwwQ1UQ/38

Reply:
We need to provide values for `state1` and `state2`. We could just assign them to `false.B` and `true.B`, or we could couple this example with Chisel ENum: https://www.chisel-lang.org/chisel3/docs/explanations/chisel-enum.html

Reply:
https://scastie.scala-lang.org/Shorla/o1ShdaY3RWKf0IIFwwQ1UQ/52

Reply:
Here is a good verilog example we could use to show updating a `Reg` and using `ChiselEnum` for a very common use case: https://insights.sigasi.com/manual/rules/verilog_case/

Reply:
https://scastie.scala-lang.org/Shorla/o1ShdaY3RWKf0IIFwwQ1UQ/58

Reply:
This is how far I have gone with the scastie : https://scastie.scala-lang.org/Shorla/o1ShdaY3RWKf0IIFwwQ1UQ/91
There is an error

Reply:
I think you had a misplaced `)`. Try this: https://scastie.scala-lang.org/QNIGowDkTc6y05OHvNdwwg

Reply:
Thank you. It's working now.

Reply:
To repeat I think this is the verilog we are copying here: 

> Here is a good verilog example we could use to show updating a Reg and using ChiselEnum for a very common use case: https://insights.sigasi.com/manual/rules/verilog_case/